src/machine.cpp

```diff
@@ -6034,11 +6066,14 @@ GcJclass* getDeclaringClass(Thread* t, GcClass* c)
 
 GcCallSite* resolveDynamic(Thread* t, GcInvocation* invocation)
 {
+  // This support for invokedynamic is limited to support for the LambdaMetafactory, as the method dispatch logic
+  // assumes the first three args are always as expected by that class. It doesn't do full indy support.
```

Comment:
Nitpick: I assume by 'indy' you mean invokedynamic - but that might not be entirely clear to others, especially given the lack of precedent in the codebase.  Could you expand that name?